src/main/java/competition/subsystems/turret/TurretSubsystem.java

```diff
@@ -44,20 +44,21 @@ public void turnRight() {
     }
 
     public void setPower(double power) {
-        if ((power < 0 && canTurnLeft()) || (power > 0 && canTurnRight())) {
+        if (canTurnLeft(power)) {
+            motor.simpleSet(power);
+        } else if (canTurnRight(power)) {
+            motor.simpleSet(power);
+        } else if (power == 0) {
             motor.simpleSet(power);
-        } else {
-            motor.simpleSet(0);
         }
-
     }
 
-    public boolean canTurnRight() {
-        return getCurrentAngle() <= maxAngleProp.get();
+    public boolean canTurnRight(double power) {
+        return (getCurrentAngle() >= maxAngleProp.get()) && (power < 0);
```

Comment:
for these checks you should only need
`  return getCurrentAngle() >= maxAngleProp.get();`